src/components/molecules/Visualizer/Engine/Cesium/hooks.ts

```diff
@@ -163,6 +194,185 @@ export default ({
     }
   }, [camera, engineAPI]);
 
+  const geodsic = useMemo(():
```

Comment:
Still long. Separate codes about camera limitter at line 197~374 into one custom hooks. Also there is still room to separate logic into a pure (= does not use React) function.

Also it would be nice to create a new file for camera limitter (e.g. `cameraLimitter.ts`)

```ts
function useCameraLimitter(/* ... */) {
  const geodic = useMemo(() => {
    const viewer = cesium.current?.cesiumElement;
    if (!viewer || !viewer.scene || viewer.isDestroyed() || !property?.cameraLimitter?.cameraLimitterTargetArea) {
      return undefined;
    }
    return calcCameraLimitterGeodsic(
      property?.cameraLimitter?.cameraLimitterTargetArea,
      viewer.scene.globel.ellipsoid
    );
  }, [
    property?.cameraLimitter?.cameraLimitterTargetArea,
  ]);

  // ...

  return {
    limiterDimensions,
    cameraViewOuterBoundaries,
    cameraViewBoundariesMaterial,
  }
}

function calcCameraLimitterGeodsic(area: Camera, ellipsoid = Ellipsoid.WGS84) {
  const centerPoint = Cartesian3.fromDegrees(
      area.lng,
      area.lat,
      0,
    );

    const cartographicCenterPoint = Cartographic.fromCartesian(centerPoint);
    const normal = ellipsoid.geodeticSurfaceNormal(centerPoint);
    const east = Cartesian3.normalize(
      Cartesian3.cross(Cartesian3.UNIT_Z, normal, new Cartesian3()),
      new Cartesian3(),
    );
    const north = Cartesian3.normalize(
      Cartesian3.cross(normal, east, new Cartesian3()),
      new Cartesian3(),
    );
    const geodesicVertical = new EllipsoidGeodesic(
      cartographicCenterPoint,
      Cartographic.fromCartesian(north),
      ellipsoid,
    );
    const geodesicHorizontal = new EllipsoidGeodesic(
      cartographicCenterPoint,
      Cartographic.fromCartesian(east),
      ellipsoid,
    );
    return { geodesicVertical, geodesicHorizontal };
}
```